File: employee-service-go/routes/employee_routes.go
package routes

import (
	"employee-service/controllers"
	"github.com/gin-gonic/gin"
)

// SetupEmployeeRoutes registers all employee-related routes
func SetupEmployeeRoutes(router *gin.Engine) {
	// Create a new employee controller
	employeeController := controllers.NewEmployeeController()

	// Group all employee routes under /api/employees
	employeeRoutes := router.Group("/api/employees")
	{
		// GET /api/employees - Get all employees
		employeeRoutes.GET("", employeeController.GetAllEmployees)
		
		// GET /api/employees/:id - Get an employee by ID
		employeeRoutes.GET("/:id", employeeController.GetEmployeeByID)
		
		// POST /api/employees - Create a new employee
		employeeRoutes.POST("", employeeController.CreateEmployee)
		
		// PUT /api/employees/:id - Update an employee
		employeeRoutes.PUT("/:id", employeeController.UpdateEmployee)
		
		// DELETE /api/employees/:id - Delete an employee
		employeeRoutes.DELETE("/:id", employeeController.DeleteEmployee)
	}
}

File: employee-service-nodejs/routes/employee.routes.js
const express = require('express');
const router = express.Router();
const employeeController = require('../controllers/employee.controller');

// Get all employees
router.get('/', employeeController.getAllEmployees);

// Get employee by ID
router.get('/:id', employeeController.getEmployeeById);

// Create a new employee
router.post('/', employeeController.createEmployee);

// Update an employee
router.put('/:id', employeeController.updateEmployee);

// Delete an employee
router.delete('/:id', employeeController.deleteEmployee);

module.exports = router;
